extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/DigitalIOThingHandler.java

```diff
@@ -0,0 +1,136 @@
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class DigitalIOThingHandler extends OwBaseThingHandler {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = ImmutableSet.of(THING_TYPE_DIGITALIO,
+            THING_TYPE_DIGITALIO2, THING_TYPE_DIGITALIO8);
+
+    private final Logger logger = LoggerFactory.getLogger(DigitalIOThingHandler.class);
+
+    public DigitalIOThingHandler(Thing thing,
+            @Nullable OwDynamicStateDescriptionProvider dynamicStateDescriptionProvider) {
+        super(thing, dynamicStateDescriptionProvider);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType) {
+            Integer ioChannel = Integer.valueOf(channelUID.getId().substring(channelUID.getId().length() - 1));
+            if (ioChannel != null && ioChannel < ((AbstractDigitalOwDevice) sensors.get(0)).getChannelCount()) {
+                Bridge bridge = getBridge();
+                if (bridge != null) {
+                    OwBaseBridgeHandler bridgeHandler = (OwBaseBridgeHandler) bridge.getHandler();
+                    if (bridgeHandler != null) {
+                        if (!((AbstractDigitalOwDevice) sensors.get(0)).writeChannel(bridgeHandler, ioChannel,
+                                command)) {
+                            logger.info("writing to channel {} in thing {} not permitted (input channel)", channelUID,
+                                    this.thing.getUID());
+                        }
+                    } else {
+                        logger.info("bridge handler not found");
+                    }
+                } else {
+                    logger.info("bridge not found");
+                }
+            }
+        }
+        super.handleCommand(channelUID, command);
+    }
+
+    @Override
+    public void initialize() {
+        Configuration configuration = getConfig();
+
+        super.initialize();
+
+        if (getThing().getStatus() == ThingStatus.OFFLINE) {
+            return;
+        }
+
+        if (this.thing.getThingTypeUID().equals(THING_TYPE_DIGITALIO)) {
+            sensors.add(new DS2405(sensorIds.get(0), this));
+        } else if (this.thing.getThingTypeUID().equals(THING_TYPE_DIGITALIO2)) {
+            sensors.add(new DS2406_DS2413(sensorIds.get(0), this));
+        } else if (this.thing.getThingTypeUID().equals(THING_TYPE_DIGITALIO8)) {
+            sensors.add(new DS2408(sensorIds.get(0), this));
+        }
+
+        // sensor configuration
+        try {
+            sensors.get(0).configureChannels();
+        } catch (OwException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            return;
+        }
+
+        validConfig = true;
+        refreshInterval = 10 * 1000;
+
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE);
```

Comment:
this overwrites any status set in the `super.initialized` call.

Reply:
I have made a change here. Please tell me what you think. If ok, I'll change the other occurences, too.

Reply:
@htreu, could you comment on this?